bitcoin/contrib/test.sh

```diff
@@ -80,8 +80,15 @@ cargo run --example ecdsa-psbt --features=bitcoinconsensus
 cargo run --example taproot-psbt --features=rand-std,bitcoinconsensus
 
 # Build the docs if told to (this only works with the nightly toolchain)
+if [ "$DO_DOCSRS" = true ]; then
+    # We use rustdoc so that we can check for broken links.
+    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly rustdoc --all-features -- -D rustdoc::broken-intra-doc-links -D warnings
+fi
+
+# Build the docs with a stable toolchain, in unison with the DO_DOCSRS command
+# above this checks that we feature guarded docs imports correctly.
 if [ "$DO_DOCS" = true ]; then
-    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly rustdoc --all-features -- -D rustdoc::broken-intra-doc-links -D warnings || exit 1
+    cargo +stable doc --all-features
```

Comment:
Shouldn't this have `-D warnings` as well?

Reply:
`-D warnings`  is a `cargo rustdoc` flag not a `cargo doc` flag (I learned while doing this PR).

Reply:
I think you can still put it into `RUSTDOCFLAGS`. `cargo doc` literally runs `rustdoc` inside.

Reply:
TIL, that's awesome I didn't realise. I changed both invocations to use `cargo doc` to make it uniform. While testing I noticed we do not have a `contrib/test.sh` file for `internals`, will add.